pageserver/src/tenant/timeline/span.rs

```diff
@@ -0,0 +1,26 @@
+#[cfg(not(debug_assertions))]
+#[inline]
+pub(crate) fn debug_assert_current_span_has_tenant_and_timeline_id() {}
+
+#[cfg(debug_assertions)]
+#[inline]
+pub(crate) fn debug_assert_current_span_has_tenant_and_timeline_id() {
+    use utils::tracing_span_assert;
+
+    pub static TIMELINE_ID_EXTRACTOR: once_cell::sync::Lazy<
+        tracing_span_assert::MultiNameExtractor<2>,
+    > = once_cell::sync::Lazy::new(|| {
+        tracing_span_assert::MultiNameExtractor::new("TimelineId", ["timeline_id", "timeline"])
+    });
+
+    match tracing_span_assert::check_fields_present([
+        &*crate::tenant::span::TENANT_ID_EXTRACTOR,
+        &*TIMELINE_ID_EXTRACTOR,
+    ]) {
+        Ok(()) => (),
+        Err(missing) => panic!(
+            "missing extractors: {:?}",
+            missing.into_iter().map(|e| e.name()).collect::<Vec<_>>()
+        ),
+    }
```

Comment:
```suggestion
    let fields = [
        &*crate::tenant::span::TENANT_ID_EXTRACTOR,
        &*TIMELINE_ID_EXTRACTOR,
    ];
    if let Err(missing) = check_fields_present(fields) {
        panic!(
            "missing extractors: {:?}",
            missing.into_iter().map(|e| e.name()).collect::<Vec<_>>()
        )
    }
```

Needs an additional use but since our scope is not a 5000 line file, I think it would be very much ok to have little as possible prefixes here.

Weird how clippy had not suggested this? Usually it is quite aggressive with matches having only one branch.

Reply:
Not on this PR.